scripts/chandra/download_script.py

```diff
@@ -0,0 +1,160 @@
+# the data products, use the following:
+
+# Imports
+import pyvo as vo
+import requests
+import argparse
+import h5py
+import numpy as np
+
+# CSC 2.1 TAP service
+tap = vo.dal.TAPService('http://cda.cfa.harvard.edu/csc21tap') # For CSC 2.1
+
+
+def get_source_detections_ids(args):
+
+    # Define the minimum source counts, minimum significance, and output file
+    # Recommend: min_cnts = 4000, min_sig = 40, max_theta = 1
+    # This functon will create a list of  IDs
+    min_cnts, min_sig, max_theta, output_file,file_path = args
+
+
+    qry = """
+    SELECT m.name, m.ra, m.dec, o.obsid, o.obi, o.region_id, o.src_cnts_aper_b,
+    o.flux_significance_b, o.flux_aper_b, o.theta, o.flux_bb_aper_b,
+    o.gti_mjd_obs, o.hard_hm,o.hard_hs, o.hard_ms, o.var_prob_b, 
+    o.var_index_b 
+    FROM csc21.master_source m, csc21.master_stack_assoc a, csc21.observation_source o, 
+    csc21.stack_observation_assoc b, csc21.stack_source s 
+    WHERE ((a.match_type = 'u') AND (o.flux_bb_aper_b IS NOT NULL) 
+    AND (o.src_cnts_aper_b > """+str(min_cnts)+""") AND (o.flux_significance_b > """+str(min_sig)+""") 
+    AND (o.theta < """+str(max_theta)+""")) AND (m.name = a.name) 
+    AND (s.detect_stack_id = a.detect_stack_id and s.region_id = a.region_id) 
+    AND (s.detect_stack_id = b.detect_stack_id and s.region_id = b.region_id) 
+    AND (o.obsid = b.obsid and o.obi = b.obi and o.region_id = b.region_id)
+    ORDER BY name ASC
+    """
+
+    cat = tap.search(qry)
+
+    # Convert the catalog to an astropy Table format.
+    cat = cat.to_table()
+
+    # Save the catalog to HDF5 format
+    with h5py.File(file_path+output_file+'_hdf5.hdf5', 'w') as hdf5_file:
+        for key in cat.colnames:
+            # Check if the column data type is a string
+            if cat[key].dtype.kind in ['U', 'S']:
+                # Encode Unicode string to byte string
+                encoded_strings = np.char.encode(cat[key], 'utf-8')
+                hdf5_file.create_dataset(key, data=encoded_strings)
+            else:
+                # Directly save the column as a dataset for non-string types
+                hdf5_file.create_dataset(key, data=cat[key])    
+
+    with open(file_path+output_file+'_ids.txt', 'w') as f:
+        for i,element in enumerate(cat['obsid'].data):
+            print(str(cat['obsid'].data[i])+'.'+str(cat['obi'].data[i])
+                  +'.'+str(cat['region_id'].data[i]), file = f)
+
+    return 1
+
+
+def retrieve(url, packageset, idx, file_path):
+    # This function retreives the data and saves them in tarballs
+    response = requests.get(url, params={
+        'version': 'cur',  # Current version of the CSC
+        'packageset': packageset
+    })
+    with open(file_path+f'package.{idx}.tar', 'wb') as output:
+        output.write(response.content)
+    return 1
+
+
+def main(args):
+
+    # Generate file of ids
+    get_source_detections_ids((args.min_cnts,args.min_sig,args.max_theta,args.output_file,args.file_path))
```

Comment:
```suggestion
    get_source_detections_ids(args.min_cnts, args.min_sig, args.max_theta, args.output_file, args.file_path)
```
The above suggestion is an improvement but yuou could also include the actual names of the parameters